apps/src/componentLibrary/tabs/_Tab.tsx

```diff
@@ -67,30 +90,39 @@ const _Tab: React.FunctionComponent<TabsProps> = ({
   const handleClick = useCallback(() => onClick(value), [onClick, value]);
   checkTabForErrors(isIconOnly, icon, text);
 
+  const buttonContent = renderButtonContent(
+    isIconOnly,
+    icon,
+    text,
+    iconLeft,
+    iconRight
+  );
+
+  const buttonElement = (
+    <button
+      type="button"
+      role="tab"
+      aria-selected={isSelected}
+      aria-controls={tabPanelId}
+      id={tabButtonId}
+      className={classNames(
+        isSelected && moduleStyles.selectedTab,
+        isIconOnly && moduleStyles.iconOnlyTab
+      )}
+      onClick={handleClick}
+      disabled={disabled}
+    >
+      {buttonContent}
+    </button>
+  );
+
   return (
     <li role="presentation">
-      <button
-        type="button"
-        role="tab"
-        aria-selected={isSelected}
-        aria-controls={tabPanelId}
-        id={tabButtonId}
-        className={classNames(
-          isSelected && moduleStyles.selectedTab,
-          isIconOnly && moduleStyles.iconOnlyTab
-        )}
-        onClick={handleClick}
-        disabled={disabled}
-      >
-        {isIconOnly && icon && <FontAwesomeV6Icon {...icon} />}
-        {!isIconOnly && (
-          <>
-            {iconLeft && <FontAwesomeV6Icon {...iconLeft} />}
-            {text && <span>{text}</span>}
-            {iconRight && <FontAwesomeV6Icon {...iconRight} />}
-          </>
-        )}
-      </button>
+      {tooltip ? (
+        <WithTooltip tooltipProps={tooltip}>{buttonElement}</WithTooltip>
```

Comment:
I tested locally implementing DSCO 
Tabs with a Tooltip prop in `aichat`. I observed that every once in a while, the tooltip location would shift.
I'm unsure why but just wanted to see if you had any ideas or if you can repro. Thanks!

https://github.com/code-dot-org/code-dot-org/assets/107423305/8f3974a2-24a9-4055-8b20-d467ff17ecb7

<img width="298" alt="Screenshot 2024-06-21 at 3 26 41 PM" src="https://github.com/code-dot-org/code-dot-org/assets/107423305/ea24583c-684e-4c04-85cf-95afec632906">

<img width="336" alt="Screenshot 2024-06-21 at 3 26 50 PM" src="https://github.com/code-dot-org/code-dot-org/assets/107423305/807bd612-d632-44d9-87bf-125580701d09">

Reply:
Interesting, thanks for sharing!
Looks like it's first render issue, also might be related to hovering the icon. I was fixing similar issue with first render in this PR. Can you please confirm if it only appears on fist render or hover on icon?
Definetely worth fixing, though not sure if it should be done in this PR or in an follow up PR (there's another possible rework of tooltip component ahead which might eliminate this bug)

Reply:
> Looks like it's first render issue, also might be related to hovering the icon. I was fixing similar issue with first render in this PR. Can you please confirm if it only appears on fist render or hover on icon? 

Yes, it was on first render and didn't always occur. Since I believe @markabarrett  now prefers NOT using a Tooltip in `aichat`, adding a tooltip as a prop is not as time-sensitive.

Reply:
Created a separate ticket for this: https://codedotorg.atlassian.net/browse/DESIGN2-130
Suggesting merge as is for now since aside from adding tooltip, there're also style updates to the tab in this PR.

Reply:
SGTM! Thanks for logging the bug.